### PR TITLE
Reprun 39

### PR DESCRIPTION
**Pull request by bbdaniels**
@kbjarkefur -- this is still crashing for me when I get to the test with `clear all`, but it solves #39 with the `checksum` implementation you suggested. I actually just replaced the whole `datasignature` functionality with that as I believe it is what people actually expect. Please check and merge if OK.

### CODE REVIEW

File: src/ado/reprun_dataline.ado [hunk 1]
```diff
@@ -33,28 +33,20 @@ cap program drop   reprun_dataline
...
-      if `run' == 2 {
-        local output = subinstr(`"`output'"',"run2","run1",.)
-        cap cf _all using "`output'/`data'.dta"
-        local srngcheck = _rc
-      }
+
+      * Handle data state
+      local output = subinstr(`"`datatmp'"',".txt",".csv",.)
+        cap export delimited using "`output'.dta" , replace
+        cap qui checksum "`output'.dta"
```
Comment (kbjarkefur) [inline src/ado/reprun_dataline.ado:43]:
```suggestion
        cap export delimited using "`output'" , replace
        cap qui checksum "`output'"
```

99% sure this was just left from previous version, but editing in suggestion to double check with you

File: src/ado/reprun_dataline.ado [hunk 2]
```diff
@@ -33,28 +33,20 @@ cap program drop   reprun_dataline
       *Get the states to be checked
       local srng "`c(sortrngstate)'"
       datasignature
-      local dsig "`r(datasignature)'"
+      local dsum "`r(datasignature)'"
```
Comment (kbjarkefur) [inline src/ado/reprun_dataline.ado:36]: If we always replace this by the checksum below, why do we even keep this?

File: src/ado/reprun.ado [hunk 3]
```diff
@@ -927,7 +933,7 @@ end
     local sl "{c |}{col `c1'}"
     local sl "`sl'{c |}{dup 6: }Seed RNG State{col `c2'}"
     local sl "`sl'{c |}{dup 6: }Sort Order RNG{col `c3'}"
-    local sl "`sl'{c |}{dup 6: }Data Signature{col `c4'}"
+    local sl "`sl'{c |}{dup 6: }Data Checksum{col `c4'}"
```
Comment (kbjarkefur) [inline src/ado/reprun.ado:936]: If we are replacing data signature with data checksum, what does the column sort order RNG add?

File: src/ado/reprun_dataline.ado (hunk 1 above)
Comment (bbdaniels) [reply src/ado/reprun_dataline.ado:43]: Honestly this could be a tempfile

File: src/ado/reprun_dataline.ado (hunk 2 above)
Comment (bbdaniels) [reply src/ado/reprun_dataline.ado:36]: Good point

File: src/ado/reprun_dataline.ado [hunk 4]
```diff
@@ -33,28 +33,20 @@ cap program drop   reprun_dataline
...
-        local srngcheck = _rc
-      }
+
+      * Handle data state
+      local output = subinstr(`"`datatmp'"',".txt",".csv",.)
+        cap export delimited using "`output'.dta" , replace
+        cap qui checksum "`output'.dta"
+        cap local srngcheck = `r(checksum)'
+          if _rc local srngcheck = 0
+          local dsum "`srngcheck'"
```
[the remark this answers is not in the document]
Comment (bbdaniels) [reply src/ado/reprun_dataline.ado:46]: I think it is more in line with what people expect, but I wanted to know if you agreed!

File: src/ado/reprun.ado (hunk 3 above)
Comment (bbdaniels) [reply src/ado/reprun.ado:936]: Tells you what the problem is

**Pull request merged**